pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BaseImmutableDictionary.java

```diff
@@ -81,6 +80,31 @@ public int indexOf(String stringValue) {
     return (index >= 0) ? index : NULL_VALUE_INDEX;
   }
 
+  @Override
+  public IntSet getDictIdsInRange(String lower, String upper, boolean includeLower, boolean includeUpper) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int compare(int dictId1, int dictId2) {
+    return Integer.compare(dictId1, dictId2);
+  }
+
+  @Override
+  public Comparable getMinVal() {
+    return (Comparable) get(0);
+  }
+
+  @Override
+  public Comparable getMaxVal() {
+    return (Comparable) get(_length - 1);
+  }
+
+  @Override
+  public Object getSortedValues() {
+    throw new UnsupportedOperationException();
```

Comment:
why is this unsupported? since it returns true for `isSorted `

Reply:
If you want a default impl, then perhaps just remove it, so the subclass must implement it.

Reply:
Added the note: This method is for the stats collection phase when sealing the consuming segment, so not required for regular immutable dictionary within the immutable segment.